usecases/schema/class.go

```diff
@@ -151,61 +150,33 @@ func (h *Handler) UpdateClass(ctx context.Context, principal *models.Principal,
 	className string, updated *models.Class,
 ) error {
 	err := h.Authorizer.Authorize(principal, "update", "schema/objects")
-	if err != nil {
-		return err
-	}
-
-	initial := h.metaReader.ReadOnlyClass(className)
-	if initial == nil {
-		return ErrNotFound
-	}
-	mtEnabled, err := validateUpdatingMT(initial, updated)
-	if err != nil {
+	if err != nil || updated == nil {
 		return err
 	}
 
 	// make sure unset optionals on 'updated' don't lead to an error, as all
 	// optionals would have been set with defaults on the initial already
 	h.setClassDefaults(updated)
+	initial := h.metaReader.ReadOnlyClass(className)
 
-	if err := h.validateImmutableFields(initial, updated); err != nil {
-		return err
-	}
-	if err := h.parser.ParseClass(updated); err != nil {
-		return err
-	}
-
-	if err := h.validator.ValidateVectorIndexConfigUpdate(ctx,
-		initial.VectorIndexConfig.(schemaConfig.VectorIndexConfig),
-		updated.VectorIndexConfig.(schemaConfig.VectorIndexConfig)); err != nil {
-		return errors.Wrap(err, "vector index config")
-	}
-
-	if err := h.validator.ValidateInvertedIndexConfigUpdate(ctx,
-		initial.InvertedIndexConfig, updated.InvertedIndexConfig); err != nil {
-		return errors.Wrap(err, "inverted index config")
-	}
-	if err := validateShardingConfig(initial, updated, mtEnabled, h.clusterState); err != nil {
-		return fmt.Errorf("validate sharding config: %w", err)
-	}
+	// first layer of defense is basic validation if class already exits
+	if initial != nil {
```

Comment:
**suggestion:** 
we could return early so we wouldn't need to scope the validators , or is it changed for specific reason  ?
```go
if initial == nil {
  return ErrNotFound
}
```

Reply:
No because of eventual consistency
At the time of request the node might not have received the updates